src/Sdk/Validation/Validator.php

```diff
@@ -0,0 +1,175 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\SdkBlueprint\Sdk\Validation;
+
+use LoyaltyCorp\SdkBlueprint\Sdk\DataTransferObject;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\InvalidRulesException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\UndefinedValidationRuleException;
+
+class Validator
+{
+    /**
+     * Repository validation rules
+     *
+     * @var mixed[]
+     */
+    public $rules = [];
+
+    /**
+     * Validate a repository
+     *
+     * @param mixed[] $data The data to validate
+     *
+     * @return string[]
+     */
+    private function validateAttributeValues(array $data): array
+    {
+        // Reset errors
```

Comment:
|`line 69`|
|---|
Just details :P but your variable is `$embeddedObjects` but your method name is `embedObjects()`, I think the method should be `getEmbeddedObjects(): array` to be "english correct" and to indicate that it is a getter.